src/main/java/teammates/ui/webapi/action/EnrollStudentsAction.java

```diff
@@ -0,0 +1,122 @@
+package teammates.ui.webapi.action;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.datatransfer.attributes.StudentAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidHttpRequestBodyException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.StudentsData;
+import teammates.ui.webapi.request.StudentsEnrollRequest;
+
+/**
+ * Enroll a list of students.
+ *
+ * <p>Create the students who are not in the course.
+ *
+ * <p>Update the students who are already existed.
+ *
+ * <p>Return all students who are successfully enrolled.
+ */
+public class EnrollStudentsAction extends Action {
+
+    private static final String ERROR_MESSAGE_SAME_TEAM_IN_DIFFERENT_SECTION =
+            "Team \"%s\" is detected in both Section \"%s\" and Section \"%s\"."
+                    + " Please use different team names in different sections";
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return authType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        if (!userInfo.isInstructor) {
+            throw new UnauthorizedAccessException("Instructor privilege is required to access this resource.");
+        }
+        String courseId = getRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_STUDENT);
+    }
+
+    @Override
+    public JsonResult execute() {
+
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        StudentsEnrollRequest enrollRequests = getAndValidateRequestBody(StudentsEnrollRequest.class);
+        List<StudentAttributes> studentsToEnroll = new ArrayList<>();
+        enrollRequests.getStudentEnrollRequests().forEach(studentEnrollRequest -> {
+            studentsToEnroll.add(StudentAttributes.builder(courseId, studentEnrollRequest.getEmail())
+                    .withName(studentEnrollRequest.getName())
+                    .withSectionName(studentEnrollRequest.getSection())
+                    .withTeamName(studentEnrollRequest.getTeam())
+                    .withComment(studentEnrollRequest.getComments())
+                    .build());
+        });
+
+        List<StudentAttributes> existingStudents = logic.getStudentsForCourse(courseId);
+        validateTeamName(existingStudents, studentsToEnroll);
```

Comment:
So In here if one of the students to enroll is not valid, the entire request will fail?

Reply:
Yes, because this error makes this student conflict with other students. If the invalid student does not affect others (e.g. the email does not match regex), it will not stop the entire request